parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,331 @@
+/**
+ *   Copyright 2016 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_PARSER_PARSE_BLOCK_PROPERTIES_HPP_
+#define QUICKSTEP_PARSER_PARSE_BLOCK_PROPERTIES_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parser/ParseLiteralValue.hpp"
+#include "parser/ParseString.hpp"
+#include "parser/ParseTreeNode.hpp"
+#include "utility/Macros.hpp"
+#include "utility/PtrList.hpp"
+
+namespace quickstep {
+
+/** \addtogroup Parser
+ *  @{
+ */
+
+/**
+ * @brief Parsed BlockPropertyItem which simply a key (property) and the
+ *        corresponding value.
+ * @details The value may actually be a list of Parse strings. This is to cover
+ *          the case where a user specifies a subset of columns to compress
+ *          rather than using the keyword 'ALL'.
+ */
+class ParseBlockPropertyItem : public ParseTreeNode {
+ public:
+  enum class Property {
+    kBlockSize,
+    kCompress,
+    kSort,
+    kType
+  };
+
+  /**
+   * @brief Creates a BlockSize block property item.
+   *
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   * @param numeric_value The size value of the block in Mb.
```

Comment:
"Mb" (with lowercase "b") actually means megabits, i.e. 128 KB. Better to write out megabytes (or mebibytes) for clarity.

Also, just for clarity, are we measuring the size in megabytes or in slots (currently 2MB each)? If the former, how do we deal with a requested size that's not an exact multiple of the slot size?